src/drive/web/modules/views/Sharings/FilesViewerSharings.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react'
+import { useQuery } from 'cozy-client'
+import { useRouter } from 'drive/lib/RouterContext'
+import { buildSharingsQuery } from 'drive/web/modules/queries'
+import { getCurrentFileId } from 'drive/web/modules/selectors'
+import { connect } from 'react-redux'
+import FilesViewer, {
+  FilesViewerLoading
+} from 'drive/web/modules/viewer/FilesViewerV2'
+import SharedDocuments from 'cozy-sharing/dist/components/SharedDocuments'
+
+const FilesViewerWithQuery = ({ sharedDocumentIds, ...props }) => {
+  const filesQuery = buildSharingsQuery(sharedDocumentIds)
+  const results = useQuery(filesQuery.definition, filesQuery.options)
+  const { router } = useRouter()
+  if (results.data) {
+    const viewableFiles = results.data.filter(f => f.type !== 'directory')
```

Comment:
I think we do not need the filter since the query is already on the files.

Reply:
This particular query is on both files and folders, so we do need it :)

Reply:
nit: I think what confused me was the variable named `filesQuery`, it could maybe be renamed to sharingQuery ?

Reply:
Well it's also used in the recent view, and it's in the sense of the `files` doctype, so I do think the name is good as it is :)

Reply:
OK